tutorial/consuming_packages/getting_started/conanfile_py/conanfile.py

```diff
@@ -1,17 +1,35 @@
+import os
+
 from conan import ConanFile
 from conan.tools.cmake import cmake_layout
+from conan.errors import ConanInvalidConfiguration
 
-
-class compressorRecipe(ConanFile):
+class CompressorRecipe(ConanFile):
     # Binary configuration
     settings = "os", "compiler", "build_type", "arch"
     generators = "CMakeToolchain", "CMakeDeps"
 
+    def validate(self):
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            raise ConanInvalidConfiguration("ARM v8 not supported")
+
     def requirements(self):
         self.requires("zlib/1.2.11")
+        # Use the system's CMake for Windows
+        # and add base64 dependency
+        if self.settings.os == "Windows":
+            self.requires("base64/0.4.0")
+        else:
+            self.tool_requires("cmake/3.19.8")
```

Comment:
This if-else reads a bit ilogical. If a conditional over cmake is desired to explain something (I would probably simplify to just 1 conditional requirement), I would do
```suggestion
        if self.settings.os != "Windows":  # we need cmake 3.19 in other platforms
            self.tool_requires("cmake/3.19.8")
```